File: GISExercise-Python/mapfunctions.py
__author__ = 'jbeaver'

import math

class MapFunctions():

    # normalize a longitude that falls outside of range(-180,180) e.g: -220
    def norm_longitude_mod(self, lon_value):
        # determine modulus of given longitude value when divided by 360 degrees
        lon_mod = lon_value % 360
        # handle the cases where value is greater than 180 degrees and smaller than -180 degrees
        if lon_mod > 180:
            diff = lon_mod - 180
            lon_mod = -180 + diff
        elif lon_mod < -180:
            diff = lon_mod + 180
            lon_mod = 180 + diff
        return lon_mod

File: GISExercise-WPF/GISiExercise_WPF/Model/MapFunctions.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace GISiExercise_WPF.Model
{
    class MapFunctions
    {
        /// <summary>
        /// Normalize a given longitude value if it is greater than 180 degrees or less than -180 degrees
        /// </summary>
        /// <param name="lon_value">Longitude value</param>
        /// <returns>Normalized longitude value</returns>
        public double normalizeLon(double lon_value)
        {
            double lon_mod, diff;
            // determine modulus of given longitude value when divided by 360 degrees
            lon_mod = lon_value % 360;
            // handle the cases where value is greater than 180 degrees and smaller than -180 degrees
            if(lon_mod > 180.0){
                diff = lon_mod - 180.0;
                lon_mod = -180.0 + diff;
            }
            else if(lon_mod < -180.0){
                diff = lon_mod + 180.0;
                lon_mod = 180.0 + diff;
            }
            return lon_mod;
        }
    }
}
